### ISSUE TITLE
MP: Identification of a copter and internal storage of copter data

### ISSUE THREAD

**Issue by patschn** [ArduPilot/MissionPlanner#1470]
Is there a way in the sourcecode of MissionPlanner to identify and to store values for a particular copter, as I don't want to access the memory of the copter directly. 

For example: I want to set values for a function, that are specific for individual copters, so the values can be loaded automatically if the copter has been connected to the pc.

Thank you in advance!

**Comment by magicrub** [comment]
I don't think I understand your request..

**Comment by patschn** [comment]
For example:
Let's say I have implemented a function, that fills a table with specifications of a copter, like batteryvoltage, batterycapacity, maximum/average speed etc. after the copter has been connected to the pc.
But in order to achieve this, I need a way to identify this copter (with an identification string or similar).

Therefore a way is required (in the source of MissionPlanner) to access this identification string and read it. Also this read data (batteryvoltage,capacity and so on) should be stored somewhere. What I discovered is Settings.Instance()['CustomString'] = 'YourValue'.
But is there something like Settings.Instance... where I can store copter specific data?

Simple:
A function reads the identification string of a connected copter (GetConnectedCopterID() returns "Copter: 1234").
Lookup coptervalues in storage. (LookupValuesInStorage("Copter: 1234") returns: (3.74V, 350mAh,...)

Does this clear up my request?
